include/aspect/newton.h

```diff
@@ -105,6 +105,69 @@ namespace aspect
        */
       void set_newton_derivative_scaling_factor(const double newton_derivative_scaling_factor);
 
+      /**
+       * Get the stabilization type used in the preconditioner.
+       */
+      std::string get_preconditioner_stabilization() const;
+
+      /**
+       * Set the stabilization type used in the preconditioner.
+       */
+      void set_preconditioner_stabilization(const std::string preconditioner_stabilization);
```

Comment:
I see what you mean by using these strings. I think it would indeed be cleaner if you defined `enum`s here and in similar places that these functions could take/return